tests/test_config.py

```diff
@@ -66,3 +66,29 @@ def test_ssl_config(certfile_and_keyfile):
     config.load()
 
     assert config.is_ssl is True
+
+
+async def asgi3app(scope, receive, send):
+    pass
+
+
+def asgi2app(scope):
+    async def asgi(receive, send):
+        raise RuntimeError("Something went wrong")
+
+    return asgi
+
+
+asgi_scope_data = [
+    (asgi3app, "asgi3", {"asgi": {"version": "3.0", "spec_version": "2.1"}}),
+    (asgi2app, "asgi2", {"asgi": {"version": "2.0", "spec_version": "2.1"}}),
+]
+
+
+@pytest.mark.parametrize(
+    "asgi2or3_app, expected_interface, expected_scope", asgi_scope_data
+)
+def test_interface_config(asgi2or3_app, expected_interface, expected_scope):
+    config = Config(app=asgi2or3_app)
+    config.load()
+    assert config.interface == expected_interface
```

Comment:
This seems to duplicate some of the testing we already do elsewhere here… Maybe we can repurpose this as a unit test for the new `.asgi_version` attribute?

```suggestion
@pytest.mark.parametrize(
    "app, asgi_version", [(asgi_app, "3.0"), (asgi2_app, "2.0")]
)
def test_interface_config(app, asgi_version):
    config = Config(app=app)
    config.load()
    assert config.asgi_version == asgi_Version
```

Reply:
ok I was making changes while you were commenting so I incorporated the test changed you want and still kept the protocol test for scopes

Reply:
that said I dont think this duplicates anything, or i failed to see any mention of interface testing, this could have been in the auto test too but there wasnt anything, but maybe I didnt search hard enough, always an option